### PR TITLE
Feat: Fetch whatsapp templates till next cursor

### PR DESCRIPTION
**Pull request by tejaswinichile**
Fixes: https://linear.app/chatwoot/issue/CW-1464/message-templates-pagination-is-missing

### CODE REVIEW

File: app/services/whatsapp/providers/whatsapp_cloud_service.rb [hunk 1]
```diff
@@ -40,8 +44,21 @@ def media_url(media_id)
...
+
+      response = HTTParty.get(response['paging']['next'])
+      save_templates(response)
+    end
+  end
+
   private
 
+  def save_templates(response)
+    whatsapp_channel.update(message_templates: response['data'], message_templates_last_updated: Time.now.utc) if response.success?
```
Comment (pranavrajs) [inline app/services/whatsapp/providers/whatsapp_cloud_service.rb:59]: Wouldn't this overwrite the previous template? I believe we should do a concat of all the responses here.

Comment (tejaswinichile) [reply app/services/whatsapp/providers/whatsapp_cloud_service.rb:60]: Yeah figured that updating the PR. 👍🏼

File: app/services/whatsapp/providers/whatsapp_cloud_service.rb [hunk 2]
```diff
@@ -40,8 +45,25 @@ def media_url(media_id)
     "#{api_base_path}/v13.0/#{media_id}"
   end
 
+  def fetch_next_templates(response)
+    loop do
+      break unless response['paging'] && response['paging']['next']
+
+      response = HTTParty.get(response['paging']['next'])
+      push_templates(response)
+    end
```
Comment (scmmishra) [inline app/services/whatsapp/providers/whatsapp_cloud_service.rb:54]:
```suggestion
    while response['paging'] && response['paging']['next']
      response = HTTParty.get(response['paging']['next'])
      push_templates(response)
    end
```

Just one minor suggestion, everything else looks great

**Review by scmmishra — APPROVED** [review]

File: app/services/whatsapp/providers/whatsapp_cloud_service.rb (hunk 2 above)
Comment (tejaswinichile) [reply app/services/whatsapp/providers/whatsapp_cloud_service.rb:54]: I added `while` before, but Rubocop gives an issue.

**Pull request merged**